driver-examples/src/main/java/com/datastax/driver/examples/json/codecs/JacksonJsonCodec.java

```diff
@@ -0,0 +1,155 @@
+ * this can be done by declaring the following dependency in your project:
+ *
+ * <p>
+ *
+ * <pre>{@code
+ * <dependency>
+ *   <groupId>com.fasterxml.jackson.core</groupId>
+ *   <artifactId>jackson-databind</artifactId>
+ *   <version>2.6.3</version>
+ * </dependency>
+ * }</pre>
+ */
+public class JacksonJsonCodec<T> implements TypeCodec<T> {
+
+  private final ObjectMapper objectMapper;
+  private final GenericType<T> javaType;
+
+  /**
+   * Creates a new instance for the provided {@code javaClass}, using a default, newly-allocated
+   * {@link ObjectMapper}.
+   *
+   * @param javaClass the Java class this codec maps to.
+   */
+  public JacksonJsonCodec(Class<T> javaClass) {
+    this(javaClass, new ObjectMapper());
+  }
+
+  /**
+   * Creates a new instance for the provided {@code javaClass}, and using the provided {@link
+   * ObjectMapper}.
+   *
+   * @param javaClass the Java class this codec maps to.
+   */
+  public JacksonJsonCodec(Class<T> javaClass, ObjectMapper objectMapper) {
+    this.javaType = GenericType.of(javaClass);
+    this.objectMapper = objectMapper;
+  }
+
+  @NonNull
+  @Override
+  public GenericType<T> getJavaType() {
+    return javaType;
+  }
+
+  @NonNull
+  @Override
+  public DataType getCqlType() {
+    return DataTypes.TEXT;
+  }
+
+  @Nullable
+  @Override
+  public ByteBuffer encode(@Nullable T value, @NonNull ProtocolVersion protocolVersion) {
+    if (value == null) return null;
+    try {
+      return ByteBuffer.wrap(objectMapper.writeValueAsBytes(value));
+    } catch (JsonProcessingException e) {
+      throw new InvalidTypeException(e.getMessage(), e);
+    }
+  }
+
+  @Nullable
+  @Override
+  public T decode(@Nullable ByteBuffer bytes, @NonNull ProtocolVersion protocolVersion) {
+    if (bytes == null) return null;
+    try {
+      return objectMapper.readValue(Bytes.getArray(bytes), toJacksonJavaType());
+    } catch (IOException e) {
+      throw new InvalidTypeException(e.getMessage(), e);
+    }
+  }
+
+  @NonNull
+  @Override
+  public String format(T value) {
+    if (value == null) return "NULL";
+    String json;
+    try {
+      json = objectMapper.writeValueAsString(value);
+    } catch (IOException e) {
+      throw new InvalidTypeException(e.getMessage(), e);
+    }
+    return Strings.quote(json);
+  }
+
+  @Nullable
+  @Override
+  @SuppressWarnings("unchecked")
+  public T parse(String value) {
+    if (value == null || value.isEmpty() || value.equalsIgnoreCase("NULL")) return null;
+    if (!Strings.isQuoted(value))
+      throw new InvalidTypeException("JSON strings must be enclosed by single quotes");
+    String json = Strings.unquote(value);
+    try {
+      return (T) objectMapper.readValue(json, toJacksonJavaType());
+    } catch (IOException e) {
+      throw new InvalidTypeException(e.getMessage(), e);
+    }
+  }
+
+  /**
+   * This method acts as a bridge between Guava's {@link
```

Comment:
Change to "This method acts as a bridge between the driver's {@link com.datastax.oss.driver.api.core.type.reflect.GenericType GenericType} API and Jackson's {@link JavaType} API.